axelor-account/src/main/java/com/axelor/apps/account/service/moveline/MoveLineServiceImpl.java

```diff
@@ -285,4 +286,12 @@ public void updatePartner(List<MoveLine> moveLineList, Partner partner, Partner
         .filter(it -> Objects.equals(it.getPartner(), previousPartner))
         .forEach(it -> it.setPartner(partner));
   }
+
+  @Override
+  public BigDecimal computeCurrencyAmount(MoveLine moveLine) {
+    BigDecimal credit = moveLine.getCredit();
+    BigDecimal debit = moveLine.getDebit();
+
+    return (debit.add(credit)).divide(moveLine.getCurrencyRate(), RoundingMode.HALF_UP);
```

Comment:
Please add a scale to `divide` and remove unnecessary parenthesis around `debit.add(credit)`